### ISSUE TITLE
`microvm` command UX overhaul

### ISSUE THREAD

**Issue by astro** [astro/microvm.nix#41]
One day I would like to rewrite the `microvm` command in Rust with improved arguments parsing. Leave your UX ideas here!

The current user interface looks like this:
```
Usage: /run/current-system/sw/bin/microvm <action> [flags]

Actions:
          -c <name>  Create a MicroVM
          -u <name>  Rebuild (update) a MicroVM
          -r <name>  Run a MicroVM in foreground
          -l         List MicroVMs

Flags:
          -f <flake> Create using another flake than git+file:///etc/nixos
          -R         Restart after update
```

**Comment by marvin** [comment]
good idea. i would suggest if u want to focus ppl on one command because ppl are lazzy then i would also suggest adding
- stop/start vm
- destroy VM
- restart/reset VM 
- also we maybe need to advance the "-u" option with a more descriptive comment
this is just my 5 cents .... 
i am willing to help, test and improve things on microVM or even on the rust development if wished.

**Comment by kraftnix** [comment]
An equivalent to `machinectl shell` for shelling / logging into a microVM would be helpful for debugging + management purposes.

**Comment by astro** [comment]
> An equivalent to `machinectl shell` for shelling / logging into a microVM would be helpful for debugging + management purposes.

That will require either one of:
- [ ] #22
- [ ] Capturing the hypervisor's/MicroVM's console to serve it on a bidirectional (UNIX) socket

**Comment by Sohalt** [comment]
It would also be nice to have a snapshot/restore commands for hypervisors, which support runtime snapshots.

**Comment by astro** [comment]
I started with a project skeleton 14 months ago but then realized: I was kinda rebuilding what [colmena](https://github.com/zhaofengli/colmena) already does - just locally instead of via ssh. Maybe we could shoehorn it into deploying microvms?

Then again, the tooling *could* look much different: it could start with a listing of VMs, asynchronously evaluating new configurations in the background, interactively offering to update them...

**Comment by alexandru0-dev** [comment]
@astro Probably that's the best way to go.
Btw can you create a server group like discord? It's probably easier to communicate that way

**Comment by astro** [comment]
@alexandru0-dev Ok, I created a chat room at https://matrix.to/#/#microvm.nix:envs.net

**Comment by shishkin** [comment]
I would like to be able to start a microvm with additional shares, like what `docker run --volume` does. I basically need interactive VMs I can use to sandbox agents to current directory but with all my tools that I can specify declaratively in nix when defining the microvm base image.

**Comment by SuperSandro2000** [comment]
That is probably out of scope and needs to go through some nix wrapper with a rebuild.
